File: Strings/notes.py
# Jameson Bultez, Strings Notes Python

# String: a data type that holds information surrounded by quotiation marks "" ''.

# Substring: a small portion of a string.

# Syntax: the conventions of a written language.

# Stupid-proofing AKA sanitization: protecting against user craziness.

# Bug: an error.

# Syntax Error: spelling was incorrect.

# Runtime Error: a user broke the program.

# Type Error: data types did not agree.

# Logic Error: programmer goofed, output will be incorrect.
    # Print debugging.
    # Start at the beginning, explaining it step by step.

# find(): gets the first instance of that text.

# concatenation: fusing strings together

# len(): gets the length of a string, including spaces.

# upper(): converts a string to ALL CAPS.

# lower(): converts a string to all lowercase.

# strip():

note = "Vienna's class"

name = input("What is your first name? \n").strip().lower().capitalize()

print("This is your name: " + name)

print(f"Hello, {name}, welcome to my program!")

sentence = "The quick brown fox jumps over the lazy dog."

print(len(sentence))

start = sentence.find("fox")
length = sentence.find("brown")
print(sentence[start : start + length])

File: Strings/notes.c
// Jameson Bultez, Strings Notes C

#include <stdio.h>
#include <string.h>

char subject[50];
char name[] = "Victoria";
char sentence[] = "The quick brown fox jumps over the lazy dog.";


int main(void){
     
    // printf("What class are you in? \n");
    // scanf("%s", subject);
    // fgets(subject, sizeof(subject), stdin);
    // printf("You are in %s. That is a cool class. \n", subject);

    // name[0] = 'T';
    // name[1] = 'o';
    // name[2] = 'r';
    // name[3] = 'i';
    // name[4] = ' ';
    // printf("%s", name);

    
    // printf("%lu \n", sizeof(sentence)); // starts counting at 0
    // printf("%d \n", strlen(sentence)); // starts counting at 1

    char one[] = "Hello ";
    char two[] = "World!";
    char three[] = "Welcome to my program. ";
    //printf("%s \n", one);

    //strcat(one, two); // only concatenate two strings at a time.
    //printf("%s \n", one);

    //strcat(three, one); // first parameter in parantheses is set to new variable.
    //printf("%s \n", three);
    char mario[] = "hello ";
    char luigi[] = "world";

    strcat(mario, luigi);
    printf("%s", mario);
    
   

    return 0;
}